vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/FileManagerTest.java

```diff
@@ -0,0 +1,143 @@
+        when(fileRepo.findById(fileId)).thenReturn(Optional.of(file));
+        when(storageEngine.deleteFile(file.getFilename(), fileId)).thenReturn(true);
+        serviceToTest.deleteFile(fileId);
+        verify(fileRepo).delete(file);
+    }
+    
+    @Test
+    public void test_storeFile_success() throws FileStorageException {
+        String fileId = "fileId";
+        VSFile file = new VSFile();
+        String relativePath = "relativePath";
+        byte[] fileBytes = fileContentString.getBytes();
+        when(storageEngine.deleteFile(file.getFilename(), null)).thenReturn(true);
+        when(fileFactory.createFile(fileName, fileId)).thenReturn(file);
+        when(storageEngine.storeFile(fileBytes, fileName, fileContentString)).thenReturn(relativePath);
+        CreationReturnValue returnValue = serviceToTest.storeFile(fileBytes, fileName, fileDescription, fileId);
+        verify(fileFactory).createFile(fileName, "text/plain");
+    }
+    
+    @Test
+    public void test_storeFile_failure() throws FileStorageException {
+        
+        String fileId = "fileId";
+        VSFile file = new VSFile();
+        byte[] fileBytes = fileContentString.getBytes();
+        when(storageEngine.deleteFile(file.getFilename(), null)).thenReturn(true);
+        when(fileFactory.createFile(fileName, "text/plain")).thenReturn(file);
+        when(storageEngine.storeFile(fileBytes, fileName, null)).thenThrow(new FileStorageException());
+        CreationReturnValue returnValue = serviceToTest.storeFile(fileBytes, fileName, fileDescription, fileId);
+
+        Assert.assertTrue(returnValue.getErrorMsgs().get(0).contains("File could not be stored: "));
+        verify(fileFactory).createFile(fileName, "text/plain");
+    }
+    
+    @Test
+    public void test_deleteFile_failure() {
+        VSFile file = new VSFile();
+        when(fileRepo.findById(fileId)).thenReturn(Optional.of(file));
+        when(storageEngine.deleteFile(file.getFilename(), "")).thenReturn(false);
+        assertFalse(serviceToTest.deleteFile(fileId));
+    }
+    
+    @Test
+    public void test_editFile_failure() {
+        VSFile file = new VSFile();
+        file.setFilename(fileName);
+        when(fileRepo.findById(fileId)).thenReturn(Optional.of(file));
+        when(storageEngine.renameFile(file.getFilename(), fileName, null)).thenReturn(true);
+        IVSFile returnedFile = serviceToTest.editFile(fileId, fileName, fileDescription);
+        assertEquals(file.getFilename(), returnedFile.getFilename());
+    }
+    @Test
+    public void test_downloadFile_failure() throws IOException {
+        String fileId = "fileId";
+        VSFile file = new VSFile();
+        when(fileRepo.findById(fileId)).thenReturn(Optional.of(file));
+        when(storageEngine.downloadFile(Mockito.any(String.class) ,Mockito.any( String.class))).thenThrow(new IOException());
+        assertThrows(IOException.class, () -> serviceToTest.downloadFile(file.getFilename(), fileId));
+    }
```

Comment:
test methods should be sorted by tested methods (all editFile tests, then all downloadFile tests, etc)